<?xml version="1.0"?>
<launch>
  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description" default="robot_description"/>
  <arg name="plugin" default="tesseract_collision/BulletDiscreteBVHManager"/>
  <arg name="monitor_links" default=""/>

  <!--
  FIRST = 0,   /**< Return at first contact for any pair of objects */
  CLOSEST = 1, /**< Return the global minimum for a pair of objects */
  ALL = 2,     /**< Return all contacts for a pair of objects */
  LIMITED = 3  /**< Return limited set of contacts for a pair of objects */
  -->
  <arg name="contact_test_type" default="2"/>
  <arg name="contact_distance"/>
  <arg name="debug" default="false"/>

  <arg name="rate" default="200"/>

  <!-- Load universal robot description format (URDF) -->
  <param name="$(arg robot_description)" command="$(find xacro)/xacro --inorder '$(find trajopt_examples)/urdf/lbr_iiwa_14_r820.xacro'" />

  <!-- The semantic description that corresponds to the URDF -->
  <param name="$(arg robot_description)_semantic" textfile="$(find trajopt_examples)/config/kuka_lbr_iiwa_14_r820.srdf" />

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="rate" value="$(arg rate)"/>
    <param name="use_gui" value="$(arg debug)" />
  </node>

  <node pkg="tesseract_monitoring" type="tesseract_monitoring_contacts_node" name="tesseract_monitoring_contacts">
    <param name="robot_description" type="string" value="$(arg robot_description)"/>
    <param name="plugin" type="string" value="$(arg plugin)"/>
    <param name="contact_distance" value="$(arg contact_distance)"/>
    <param name="monitor_links" value="$(arg monitor_links)"/>
    <param name="contact_test_type" value="$(arg contact_test_type)"/>
    <param name="publish_environment" value="$(arg debug)"/>
  </node>

</launch>
